name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SelfWealthPDFExtractor.java

```diff
@@ -0,0 +1,184 @@
+package name.abuchen.portfolio.datatransfer.pdf;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Map;
+
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.Block;
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.DocumentType;
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.Transaction;
+import name.abuchen.portfolio.model.AccountTransaction;
+import name.abuchen.portfolio.model.BuySellEntry;
+import name.abuchen.portfolio.model.Client;
+import name.abuchen.portfolio.model.PortfolioTransaction;
+import name.abuchen.portfolio.model.Transaction.Unit;
+import name.abuchen.portfolio.money.Money;
+
+@SuppressWarnings("nls")
+public class SelfWealthPDFExtractor extends AbstractPDFExtractor
+{
+    public SelfWealthPDFExtractor(Client client)
+    {
+        super(client);
+
+        addBankIdentifier("SelfWealth"); //$NON-NLS-1$
+
+        addBuySellTransaction();
+    }
+
+    @Override
+    public String getPDFAuthor()
+    {
+        return ""; //$NON-NLS-1$
+    }
+
+    @Override
+    public String getLabel()
+    {
+        return "SelfWealth"; //$NON-NLS-1$
+    }
+
+    private void addBuySellTransaction()
+    {
+        DocumentType type = new DocumentType("(Buy|Sell) Confirmation");
+        this.addDocumentTyp(type);
+
+        Transaction<BuySellEntry> pdfTransaction = new Transaction<>();
+        pdfTransaction.subject(() -> {
+            BuySellEntry entry = new BuySellEntry();
+            entry.setType(PortfolioTransaction.Type.BUY);
+            return entry;
+        });
+
+        Block firstRelevantLine = new Block("^(Buy|Sell) Confirmation$");
+        type.addBlock(firstRelevantLine);
+        firstRelevantLine.set(pdfTransaction);
+
+        pdfTransaction
+                // Is type --> "Sell" change from BUY to SELL
+                .section("type").optional()
+                .match("^(?<type>Sell) Confirmation$")
+                .assign((t, v) -> {
+                    if (v.get("type").equals("Sell"))
+                    {
+                        t.setType(PortfolioTransaction.Type.SELL);
+                    }
+                })
+
+                // JOHN DOE A/C Reference No: T20210701123456­-1
+                .section("note")
+                .match(" Reference No: (?<note>.*)$")
```

Comment:
a note is everytime optional...
so change the line 
`.section("note")` to `.section("note").optional()`

change the line to
`.match("^.* Reference No: (?<note>.*)$")`